api/src/test/java/org/pmiops/workbench/db/dao/InstitutionDaoTest.java

```diff
@@ -3,65 +3,175 @@
 import static com.google.common.truth.Truth.assertThat;
 import static com.google.common.truth.Truth8.assertThat;
 
-import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.pmiops.workbench.db.model.DbInstitution;
+import org.pmiops.workbench.db.model.DbInstitutionEmailAddress;
+import org.pmiops.workbench.db.model.DbInstitutionEmailDomain;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
 import org.springframework.dao.DataIntegrityViolationException;
 import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
 import org.springframework.test.context.junit4.SpringRunner;
 
 @RunWith(SpringRunner.class)
 @DataJpaTest
-@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+@DirtiesContext(classMode = ClassMode.BEFORE_EACH_TEST_METHOD)
 public class InstitutionDaoTest {
+
   @Autowired InstitutionDao institutionDao;
 
-  private DbInstitution testInst = new DbInstitution("Broad", "The Broad Institute");
+  private DbInstitution institutionNoEmailAddressesOrDomains;
+  private DbInstitution institutionWithEmailAddressesAndDomains;
+  private Set<DbInstitutionEmailAddress> emailAddresses;
+  private Set<DbInstitutionEmailDomain> emailDomains;
 
   @Before
   public void setUp() {
-    testInst = institutionDao.save(testInst);
+    institutionNoEmailAddressesOrDomains =
+        institutionDao.save(
+            new DbInstitution().shortName("Broad").displayName("The Broad Institute"));
+
+    emailAddresses =
+        Stream.of(new DbInstitutionEmailAddress().setEmailAddress("emailAddress"))
+            .collect(Collectors.toCollection(HashSet::new));
+    emailDomains =
+        Stream.of(new DbInstitutionEmailDomain().setEmailDomain("emailDomain"))
+            .collect(Collectors.toCollection(HashSet::new));
+    institutionWithEmailAddressesAndDomains =
+        institutionDao.save(
+            new DbInstitution()
+                .shortName("Broad1")
+                .displayName("The Broad Institute")
+                .emailAddresses(emailAddresses)
+                .emailDomains(emailDomains));
   }
 
   @Test
   public void test_save() {
-    assertThat(institutionDao.findAll()).hasSize(1);
-    institutionDao.save(new DbInstitution("VUMC", "Vanderbilt"));
-    assertThat(institutionDao.findAll()).hasSize(2);
+    // no need to call save since we have a setup method.
```

Comment:
code under test shouldn't really be executed in setup() if we can avoid it. It's just for mocks and fixtures.  You'll still fail your tests if it blows up, but you should ideally have assertions after important steps, especially in a test_save() method.

Reply:
updated